[ADD] ci: enable tests using GitHub actions t#54296

luisg123v opened the pull request:
The following tests are added:
- pre-commit-vauxoo
- No dependency files
- Odoo

luisg123v commented:
@moylop260 

Could you review, please?

I know it's red, but that's why I wanted to enable it. Next step is to fix those reds.

Regards,

moylop260 commented on .github/workflows/pre-commit.yml:
```diff
@@ -0,0 +1,33 @@
+        name: [mandatory, optional]
+    steps:
+      - uses: actions/checkout@v2
+      - name: Upgrade pre-commit-vauxoo to te latest version
```
```suggestion
      - name: Upgrade pre-commit-vauxoo to the latest version
```

moylop260 commented on .github/workflows/pre-commit.yml:
```diff
@@ -0,0 +1,33 @@
+      - name: Upgrade pre-commit-vauxoo to te latest version
+        run: pip3 install --upgrade pre-commit-vauxoo
+      - name: Run pre-commit-vauxoo
+        run: pre-commit-vauxoo
```
How are you set optional or mandatory to the command?

luisg123v replied:
Setting variable from job name:

```
env:
  PRECOMMIT_HOOKS_TYPE: ${{ matrix.name }}
```

moylop260 reviewed (commented):
The OPTIONAL pipeline is showing green status

Shouldn't be yellow one?

moylop260 commented:
Could you merge the build pre-commit-optional and pre-commit-mandatory in order to review all the logs only in one build even if it is green, please?

Since github doesn't have yellow status or allow_failure as gitlab so the optional will be green all the time but we have 2 pre-commit builds so the developer will need to check the log of optional but they could get confused about what pre-commit build check

IMHO it is better if you have the full logs only in one build for this case in order to check only one output instead of 2

luisg123v commented:
@moylop260 I agree, I wasn't aware GitHub doesn't support yellow when I implemented this

moylop260 commented:
excellent!

Also, you even don't need the container to run the tests

I think it could be even faster running over the original GH ubuntu but with cache

I will make a PR to your PR...

moylop260 commented:
Please, check the following changes:
 - https://github.com/Vauxoo/addons-vauxoo/pull/1564

It is running the pre-commit build spending 23s

moylop260 commented:
Also, could you add a error if the dev adds `oca_dependencies.txt` file in the CI, please?

luisg123v committed:
[ADD] ci: enable tests using GitHub actions

The following tests are added:
- pre-commit-vauxoo
- No dependency files
- Odoo

moylop260 commented:
👍 
(After removing the configuration files)
